SlevomatCodingStandard/Helpers/ClassHelper.php

```diff
@@ -107,6 +110,30 @@ public static function getTraitUsePointers(File $phpcsFile, int $classPointer):
 		return $useStatements;
 	}
 
+	public static function getPropertiesCount(File $file, int $position): int
+	{
+		return count(self::getProperties($file, $position));
+	}
+
+	/**
+	 * @param File $file
+	 * @param int $position
+	 * @return int[]
+	 */
+	public static function getProperties(File $file, int $position): array
+	{
+		$tokens = $file->getTokens();
+		$token = $tokens[$position];
+		$pointer = $token['scope_opener'];
+
+		return array_filter(
+			TokenHelper::findNextAll($file, T_VARIABLE, $pointer + 1, $token['scope_closer']),
+			static function ($variablePointer) use ($file): bool {
+				return PropertyHelper::isProperty($file, $variablePointer);
+			}
+		);
```

Comment:
I discovered that this method (the current as well as the previous implementation) does not take the possibility into account that this class itself might define anonymous classes with their own properties. Currently these properties are added to the list of class properties, which I think is incorrect. As an example, in the following class, the number of class properties is counted as 11:

```php
<?php

class FineAmountOfPropertiesWithNestedPropertiesClass
{
	protected $property1;
	protected $property2;
	protected $property3;
	protected $property4;
	protected $property5;
	protected $property6;
	protected $property7;
	protected $property8;
	protected $property9;
	protected $property10;

	protected function anonymousClassMethod()
	{
		return new class {
			protected $anonymousProperty1;
		};
	}
}
```

To resolve this, you can use the `level` of a token: all property tokens should be exactly one level higher than the level of the scope opener token.

I tried to first this into the above implementation above, but the closure started to work against me.

So I ended up with the following:

```php
		$tokens = $file->getTokens();
		$token = $tokens[$position];
		$scopeOpenerPointer = $token['scope_opener'];
		$scopeCloserPointer = $token['scope_closer'];
		$acceptedLevel = $tokens[$scopeOpenerPointer]['level'] + 1;

		$propertyPointers = [];
		foreach (TokenHelper::findNextAll($file, T_VARIABLE, $scopeOpenerPointer + 1, $scopeCloserPointer) as $variablePointer) {
			if (($tokens[$variablePointer]['level'] === $acceptedLevel) && PropertyHelper::isProperty($file, $variablePointer)) {
				$propertyPointers[] = $variablePointer;
			}
		}
		return $propertyPointers;
```